execute/plugin_functions.go

```diff
@@ -416,6 +500,55 @@ func mergeCommitObservations(
 	return results, nil
 }
 
+func mergeMessageHashes(
```

Comment:
I didn't review these merge functions :(

Somehow the commit versions are significantly simpler, I haven't been able to wrap my head around why.

Reply:
It took me some time to wrap my head around it as well 😅